src/functional/fencil_processors/runners/gtfn_cpu.py

```diff
@@ -13,36 +13,56 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 
-from typing import Any
+from dataclasses import dataclass, field
+from typing import Any, Optional
 
 import numpy
 
-from functional.fencil_processors.builders import cpp as cpp_callable
-from functional.fencil_processors.codegens.gtfn import gtfn_module as gtfn_codegen
-from functional.fencil_processors.processor_interface import fencil_executor
+from functional.fencil_processors.builders import cache  # , cpp as cpp_callable
+from functional.fencil_processors.builders.cpp import bindings
+from functional.fencil_processors.builders.cpp.build import CMakeProject
+from functional.fencil_processors.codegens.gtfn import gtfn_module
+from functional.fencil_processors.processor_interface import FencilExecutor, FencilProcessorProtocol
+from functional.fencil_processors.source_modules.cpp_gen import CPP_DEFAULT, CppLanguage
 from functional.iterator import ir
 
 
 def convert_arg(arg: Any) -> Any:
     view = numpy.asarray(arg)
     if view.ndim > 0:
-        return memoryview(view)
+        return memoryview(view)  # type: ignore[arg-type] # mypy seems unaware that ndarray is compatible with buffer protocol
     else:
         return arg
 
 
-# TODO(ricoh): change style to declarative pipeline
-@fencil_executor
-def run_gtfn(itir: ir.FencilDefinition, *args, **kwargs):
-    """
-    Execute the iterator IR fencil with the provided arguments.
+@dataclass(frozen=True)
+class GTFNExecutor(FencilExecutor):
```

Comment:
Why turn this into a class? This comes from a discussion with @petiaccja a while back, where we thought of customizing fencil processors by generating them with customization, rather than passing configuration through the keyword args.

In this case I chose a class, rather than a function generating a function because I suspect it is more readable to the uninitiated. Also it allows naming the configured processor, which helps with test fixture ids.